tests/test_maudau.py

```diff
@@ -0,0 +1,127 @@
+import datetime as DT
+import json
+import os
+import pytest
+import tempfile
+from pyspark.sql import SparkSession
+from pyspark.sql.types import StructField, StructType, StringType
+from mozetl.maudau import maudau as M
+
+NARROW_SCHEMA = StructType([
+    StructField("client_id",             StringType(),  True),
+    StructField("submission_date_s3",    StringType(),  False),
+    StructField("subsession_start_date", StringType(), True)])
+
+generated = M.format_as_submission_date(M.today.isoformat())
+
+spark = (
+    SparkSession
+    .builder
+    .master("local")
+    # Uncomment to run while offline: from pyspark.conf import SparkConf
+    # .config(conf=SparkConf().set("spark.driver.host", "localhost"))
+    .appName("maudau")
+    .getOrCreate()
+)
+
+
+@pytest.fixture
+def make_frame():
+    cols = ['client_id', 'subsession_start_date', 'submission_date_s3']
+    values = [
+        ('a', '2017-05-01', '20170508'),
+        ('b', '2017-05-01', '20170501'),
+        ('c', '2017-05-01', '20170510'),
+        ('a', '2017-05-02', '20170503'),
+        ('b', '2017-05-03', '20170503'),
+        ('b', '2017-05-04', '20170511'),
+        ('a', '2017-05-05', '20170507'),
+    ]
+    rows = map(json.dumps, [dict(zip(cols, tup)) for tup in values])
+    jrdd = spark.sparkContext.parallelize(rows)
+    return spark.read.json(jrdd, NARROW_SCHEMA)
+
+
+def test_generate_filter_parameters():
+    expected0 = (
+        '2017-01-31', '2017-02-01', '20170131', '20170210'
+    )
+    actual0 = M.generate_filter_parameters(DT.date(2017, 1, 31), 0)
+    assert expected0 == actual0, str(actual0)
+
+    expected28 = (
+        '2017-01-03', '2017-02-01', '20170103', '20170210'
+    )
+    actual28 = M.generate_filter_parameters(DT.date(2017, 1, 31), 28)
+    assert expected28 == actual28
+
+
+def test_generate_counts():
+    frame = make_frame()
+    since = '20170501'
+    counts = M.generate_counts(frame, since, '20170506')
+    expected = {'day': '20170505', 'dau': 1, 'mau': 3,
+                'generated_on': generated}
+    assert counts[-1] == expected, str(counts[-1])
+    counts2 = M.generate_counts(frame, since, '20170504')
+    expected2 = {'day': '20170503', 'dau': 1, 'mau': 3,
+                 'generated_on': generated}
+    assert counts2[-1] == expected2, str(counts2[-1])
+
+
+def test_parse_last_rollup():
+    tempdir = tempfile.mkdtemp()
+    cwd = os.getcwd()
+    basename = "engagement_ratio.csv".format(generated)
+    data = '\r\n'.join([
+        "day,dau,mau,generated_on",
+        "20170503,1,3,20170620",
+        "20170504,1,3,20170620",
+        ""
+        ])
+    expected = (
+        '20170504',
+        [{'dau': '1', 'mau': '3', 'day': '20170503',
+          'generated_on': '20170620'}]
+    )
+    try:
```

Comment:
A few options -- the first is to create a test fixture that sets up the directory and cleans up appropriately after the test, the second is to just change the directory at the end in the normal function state.

You could also just use relative paths from the temporary directory and not worry about changing the working path.